Core/src/org/sleuthkit/autopsy/datamodel/BlackboardArtifactNode.java

```diff
@@ -340,6 +361,20 @@ private static Lookup createLookup(BlackboardArtifact artifact) {
         }
     }
 
+    private static Lookup createLookup(BlackboardArtifactNodeKey nodeKey) {
+        Content content;
+        if (nodeKey.isLookupIsAssociatedFile()) {
+            content = nodeKey.getAssociatedFile();
+            
+            if (content == null) {
+                return Lookups.fixed(nodeKey.getArtifact());
+            } else {
+                return Lookups.fixed(nodeKey.getArtifact(), content);
+            }
+        } else {
+            return createLookup(nodeKey.getArtifact());
```

Comment:
Now that we have two createLookups, I'd rename this one to createLookupWithSource (or something) to make it more clear that the two have different behavior and aren't just different arguments.   Or, refactor to make it it a single method.